src/librustc_codegen_ssa/mir/rvalue.rs

```diff
@@ -327,13 +327,28 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                                 if er.end != er.start
                                     && scalar.valid_range.end() > scalar.valid_range.start()
                                 {
-                                    // We want `table[e as usize]` to not
+                                    // We want `table[e as usize ± k]` to not
                                     // have bound checks, and this is the most
-                                    // convenient place to put the `assume`.
+                                    // convenient place to put the `assume`s.
                                     let ll_t_in_const =
                                         bx.cx().const_uint_big(ll_t_in, *scalar.valid_range.end());
-                                    let cmp = bx.icmp(IntPredicate::IntULE, llval, ll_t_in_const);
-                                    bx.assume(cmp);
+                                    let cmp_end =
+                                        bx.icmp(IntPredicate::IntULE, llval, ll_t_in_const);
+
+                                    if *scalar.valid_range.start() > 0 {
+                                        // While this makes the codegen test pass, this will not
+                                        // elide every bounds check due to an LLVM issue.
+                                        // See https://github.com/rust-lang/rust/issues/75525
+                                        let ll_t_in_const = bx
```

Comment:
Style nit: I would probably avoid shadowing here and name the variables in a way that represents whether they are the upper or lower bound.